src/main/java/ch/njol/skript/sections/EffSecFireworkLaunch.java

```diff
@@ -0,0 +1,166 @@
+	"\t\t\tcancel event",
+	"",
+	"command /firework:",
+	"\ttrigger:",
+	"\t\tlaunch a firework with effects ball large coloured red at player:",
+	"\t\t\tset metadata value \"cancelDamage\" of event-firework to true"})
+@Since("2.4, INSERT VERSION (with section)")
+public class EffSecFireworkLaunch extends EffectSection {
+
+	public static class FireworkLaunchEvent extends Event {
+		private final Firework firework;
+
+		public FireworkLaunchEvent(Firework firework) {
+			this.firework = firework;
+		}
+
+		public Firework getFirework() {
+			return firework;
+		}
+
+		@Override
+		public HandlerList getHandlers() {
+			throw new IllegalStateException();
+		}
+	}
+
+	static {
+		Skript.registerSection(EffSecFireworkLaunch.class, "(launch|deploy) [[a] firework [with effect[s]]] %fireworkeffects% at %locations% [([with] (duration|power)|timed) %-number%]");
+		EventValues.registerEventValue(FireworkLaunchEvent.class, Firework.class, new Getter<Firework, FireworkLaunchEvent>() {
+			@Override
+			public Firework get(FireworkLaunchEvent fireworkLaunchEvent) {
+				return fireworkLaunchEvent.getFirework();
+			}
+		}, EventValues.TIME_NOW);
+	}
+
+	@SuppressWarnings("NotNullFieldNotInitialized")
+	private Expression<FireworkEffect> effects;
+	@SuppressWarnings("NotNullFieldNotInitialized")
+	private Expression<Location> locations;
+	@Nullable
+	private Expression<Number> lifetime;
+
+	@Nullable
+	private Trigger trigger;
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs,
+						int matchedPattern,
+						Kleenean isDelayed,
+						SkriptParser.ParseResult parseResult,
+						@Nullable SectionNode sectionNode,
+						@Nullable List<TriggerItem> triggerItems) {
+		effects = (Expression<FireworkEffect>) exprs[0];
+		locations = (Expression<Location>) exprs[1];
+		lifetime = (Expression<Number>) exprs[2];
+
+		if (sectionNode != null) {
+			trigger = loadCode(sectionNode, "fireworklaunch", FireworkLaunchEvent.class);
+		}
+
+		return true;
+	}
+
+	@Override
+	@Nullable
+	@SuppressWarnings({"unchecked", "rawtypes"})
+	protected TriggerItem walk(Event e) {
+		Object localVars = Variables.copyLocalVariables(e);
+
+		Consumer<? extends Firework> consumer;
+		if (trigger != null) {
+			consumer = o -> {
+				FireworkLaunchEvent fireworkLaunchEvent = new FireworkLaunchEvent(o);
+				// Copy the local variables from the calling code to this section
+				Variables.setLocalVariables(fireworkLaunchEvent, localVars);
+				trigger.execute(fireworkLaunchEvent);
+				Variables.setLocalVariables(e, localVars); // Carry over variable changes to the rest of the main trigger
+			};
+		} else {
+			consumer = null;
+		}
+
+		Number power = lifetime != null ? lifetime.getSingle(e) : 1;
+		if (power == null)
+			power = 1;
+		for (Location location : locations.getArray(e)) {
+			Firework firework = location.getWorld().spawn(location, Firework.class, (Consumer) consumer);
```

Comment:
Why id the consumer being casted to a consumer without it's generic?

Reply:
Even instead of casting, you can just change it to `Consumer<Entity>` which will allow you to get rid of the cast and the suppressed warnings annotation

Reply:
Then the method itself would error too, because they're not of the same type (`Firework` and `Entity`)
![image](https://user-images.githubusercontent.com/29547183/179389933-d5afdd68-af88-48d9-9211-53c87e352177.png)
If you change it to `Consumer<Firework>` I think it should work without casts